app/views/shell/_top.html.erb

```diff
@@ -44,20 +44,20 @@
     <% end %>
   </head>
   <% unless internal_navigation? %>
-    <body data-user-status="<%= user_logged_in_status %>" data-pusher-key="<%= ApplicationConfig["PUSHER_KEY"] %>">
-      <div id="body-styles"></div>
-      <% if user_signed_in? %>
-        <%= render "layouts/user_config" %>
-      <% end %>
+    <% cache_key_heroku_slug("top-html-and-config--#{user_signed_in?}") %>
```

Comment:
I want to make sure I understand how this works. This will cache the user_config script but that script will still execute and choose the correct theme based on the user's preferences, is that correct? It wont cache the result of the script.

Reply:
Yes. This will cache the HTML, but the script will still execute exactly the same.

Because this code ultimately needs to be cached at the edge, it does not include any info about the user _on the server_, the customizations are actually fetched from this line...

```javascript
const bodyClass = localStorage.getItem('config_body_class');
```

Which reads what's cached in the user's browser. So no cache leaks user-to-user.

We can't use stuff like `current_user` in these parts of the app because that would indeed be leaked from one user to the next when it is served as dumb html to the next user.